pkg/transformers/refvars.go

```diff
@@ -0,0 +1,82 @@
+package transformers
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-sigs/kustomize/pkg/expansion"
+	"github.com/kubernetes-sigs/kustomize/pkg/resmap"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type refvarTransformer struct {
+	pathConfigs []PathConfig
+	vars        map[string]string
+}
+
+func NewRefVarTransformer(vars map[string]string) (Transformer, error) {
+	return &refvarTransformer{
+		vars: vars,
+		pathConfigs: []PathConfig{
+			{
+				GroupVersionKind: &schema.GroupVersionKind{Kind: "StatefulSet"},
+				Path:             []string{"spec", "template", "spec", "initContainers", "command"},
+			},
+			{
+				GroupVersionKind: &schema.GroupVersionKind{Kind: "StatefulSet"},
+				Path:             []string{"spec", "template", "spec", "containers", "command"},
+			},
+			{
+				GroupVersionKind: &schema.GroupVersionKind{Kind: "Job"},
+				Path:             []string{"spec", "template", "spec", "containers", "command"},
+			},
+		},
+	}, nil
+}
+
+func (rv *refvarTransformer) Transform(resources resmap.ResMap) error {
```

Comment:
Needs a comment.

The comment in L37 to 45 can be moved up and adapted.

Reply:
This is done